frontend/catalyst/utils/CMakeLists.txt

```diff
@@ -1,9 +1,27 @@
+project(catalyst_frontend_utils)
```

Comment:
Curious what the effect of nesting projects is in cmake :)

Reply:
The main advantage is being able to build subcomponents separately (where useful) and avoid pulling everything to the "god-level" project build. Ideally, sub project targets can be used in parent projects, and it helps to scope source/header/bins to that specific subproject, so we get much nicer granularity when it comes to using specific targets within the rest of the ecosystem.